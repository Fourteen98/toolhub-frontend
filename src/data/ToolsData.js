const ToolsData = [
  {
    id: 1,
    name: 'Depictor',
    Author: 'Damilare Adedoyin',
    description: 'A tool for creating diagrams and flowcharts',
    link: 'https://depictor.app/',
    category: 'Diagramming',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: 'https://www.draw.io/',
        api_url: '',
        api_docs_url: 'https://www.draw.io/docs/',
        for_wikis: '',
        license: 'Apache License 2.0',
      },
    ],
  },
  {
    id: 2,
    name: 'Translate',
    Author: 'Stefanova Salavina',
    description: 'A tool for translating text',
    link: 'https://translate.google.com/',
    category: 'Translation',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: '',
      },
    ],
  },
  {
    id: 3,
    name: 'QSML',
    Author: 'Silvestre Herrera',
    description: 'A tool for creating and editing QSML files',
    link: 'https://qsml.app/',
    category: 'File Editing',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'Apache License 2.0',
      },
    ],
  },
  {
    id: 4,
    name: 'TinkerCell',
    Author: 'Hongkai Ji',
    description: 'A tool for creating and editing TinkerCell files',
    link: 'https://tinkercell.com/',
    category: 'Rendering',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: '',
      },
    ],
  },
  {
    id: 5,
    name: 'Minefield',
    Author: 'Pierrick Charron',
    description: 'A tool for creating and editing Minefield files',
    link: 'https://minefield.app/',
    category: 'Web Development',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: '',
      },
    ],
  },
  {
    id: 6,
    name: 'BioModels Database',
    Author: 'James P. Overton',
    description: 'A tool for creating and editing BioModels Database files',
    link: 'https://www.ebi.ac.uk/biomodels/',
    category: 'Biology',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: '',
      },
    ],
  },
  {
    id: 7,
    name: 'Close to me',
    Author: 'Manske Jonathan',
    description: 'A tool for creating and editing Close to me files',
    link: 'https://www.closetome.app/',
    category: 'Communication',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: 'https://www.closetome.app/privacy',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: '',
      },
    ],
  },
  {
    id: 8,
    name: 'MultiSearch',
    Author: 'Harald Sack',
    description: 'A tool for creating and editing MultiSearch files',
    link: 'https://multisearch.app/',
    category: 'Search',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'MIT',
      },
    ],
  },
  {
    id: 9,
    name: 'Visualize',
    Author: 'Simon Fraser',
    description: 'A tool for creating and editing Visualize files',
    link: 'https://visualize.app/',
    category: 'Visualization',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: 'https://visualize.app/api',
        api_docs_url: '',
        for_wikis: '',
        license: 'Apache License 2.0',
      },
    ],
  },
  {
    id: 10,
    name: 'BioSimulators',
    Author: 'Magdalena Krzywinski',
    description: 'A tool for creating and editing BioSimulators files',
    link: 'https://biosimulators.org/',
    category: 'Biology',
    Annotations: [
      {
        icon: 'https://biosimulators.org/static/images/favicon.ico',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'Apache License 2.0',
      },
    ],
  },
  {
    id: 11,
    name: 'Wikidata Skim',
    Author: 'Jonathan Rees',
    description: 'A tool for creating and editing Wikidata Skim files',
    link: 'https://skim.wmflabs.org/',
    category: 'Data',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'GNU General Public License v3.0',
      },
    ],
  },
  {
    id: 12,
    name: 'Wdqs-proxy',
    Author: 'Lucas Werkmeister',
    description: 'A tool for creating and editing Wdqs-proxy files',
    link: 'https://wdqs-proxy.toolforge.org/',
    category: 'Web Tools',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: '',
      },
    ],
  },
  {
    id: 13,
    name: 'Wrdz',
    Author: 'Daniel Mietchen',
    description: 'A tool for creating and editing Wrdz files',
    link: 'https://wrdz.toolforge.org/',
    category: 'NLP',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'CC BY-SA 4.0',
      },
    ],
  },
  {
    id: 14,
    name: 'Wikidata Query Service',
    Author: 'Martin Urbanec',
    description: 'A tool for creating and editing Wikidata Query Service files',
    link: 'https://query.wikidata.org/',
    category: 'Information Retrieval',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'CC0',
      },
    ],
  },
  {
    id: 15,
    name: 'Molecular Design Toolkit',
    Author: 'Matthew P. Humphreys',
    description: 'A tool for creating and editing Molecular Design Toolkit files',
    link: 'https://molecular-design-toolkit.org/',
    category: 'Chemistry',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'MIT',
      },
    ],
  },
  {
    id: 16,
    name: 'D3',
    Author: 'Mike Bostock',
    description: 'A tool for creating and editing D3 files',
    link: 'https://d3js.org/',
    category: 'Visualization',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'BSD-3-Clause',
      },
    ],
  },
  {
    id: 17,
    name: 'D3.js',
    Author: 'Elias Zamaria',
    description: 'A tool for creating and editing D3.js files',
    link: 'https://d3js.org/',
    category: 'Design',
    Annotations: [
      {
        icon: 'https://d3js.org/favicon.ico',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'BSD-3-Clause',
      },
    ],
  },
  {
    id: 18,
    name: 'D5.js',
    Author: 'Elias Zamaria',
    description: 'A tool for creating and editing D3.js files',
    link: 'https://d3js.org/',
    category: 'Design',
    Annotations: [
      {
        icon: '',
        feedback_url: '',
        privacy_policy: '',
        replaced_by: '',
        api_url: '',
        api_docs_url: '',
        for_wikis: '',
        license: 'MIT',
      },
    ],
  },
];

export default ToolsData;
