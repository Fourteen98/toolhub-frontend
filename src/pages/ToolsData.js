const ToolsData = [
  {
    name: 'Depictor',
  },
  {
    name: 'Translate',
  },
  {
    name: 'QSML',
  },
  {
    name: 'TinkerCell',
  },
  {
    name: 'Minefield',
  },
  {
    name: 'BioModels Database',
  },
  {
    name: 'Close to me',
  },
  {
    name: 'MultiSearch',
  },
  {
    name: 'Visualize',
  },
  {
    name: 'BioSimulators',
  },
  {
    name: 'Wikidata Skim',
  },
  {
    name: 'Wdqs-proxy',
  },
  {
    name: 'Wrdz',
  },
  {
    name: 'Wikidata Query Service',
  },
  {
    name: 'Molecular Design Toolkit',
  },
  {
    name: 'D3',
  },
];

export default ToolsData;
