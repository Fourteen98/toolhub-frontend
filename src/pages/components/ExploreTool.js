import React from 'react';
import Tool2 from '../../assets/Tool2.jpg';

export default function ExploreTool({ tool }) {
  const annotationsField = tool.Annotations[0];
  console.log(annotationsField);
  // const renderAnnotations = annotationsField.map((annotation) => (
  //   <tr id={annotation.key}>
  //     <td>{annotation.key}</td>
  //     <td>{annotation.value}</td>
  //   </tr>
  // ));
  const returnNull = (field) => {
    if (field === '' || field === null) {
      return 'Missing!';
    }
    return field;
  };

  const returnClass = (field) => {
    if (field === '' || field === null) {
      return 'missing';
    }
    return '';
  };

  return (
    <div className="explore--tools--container">
      <img className="tool--card--img" src={Tool2} alt={tool.name} />
      <div className="left--container">
        <h3 className="tool--name">
          {tool.name}
        </h3>
        <p>
          <span>Description: </span>
          {tool.description}
        </p>
        <p>
          <span>Author: </span>
          {tool.Author}
        </p>
        <p>
          <span>Category: </span>
          {tool.category}
        </p>
        <p>
          <span>Source Code : </span>
          <a href={tool.link} target="_blank" rel="noreferrer"> Check here </a>
        </p>
      </div>
      <div className="right--container">
        <table className="explore--tools--tbl">
          <thead>
            <tr>
              <th>Field</th>
              <th>Value</th>
            </tr>
          </thead>
          <tbody>
            <tr className={returnClass(annotationsField.icon)}>
              <td>icon</td>
              <td>{returnNull(annotationsField.icon)}</td>
            </tr>
            <tr className={returnClass(annotationsField.feedback_url)}>
              <td>feedback_url</td>
              <td>{returnNull(annotationsField.feedback_url)}</td>
            </tr>
            <tr className={returnClass(annotationsField.license)}>
              <td>license</td>
              <td>{returnNull(annotationsField.license)}</td>
            </tr>
            <tr className={returnClass(annotationsField.privacy_policy)}>
              <td>privacy_policy</td>
              <td>{returnNull(annotationsField.privacy_policy)}</td>
            </tr>
            <tr className={returnClass(annotationsField.replaced_by)}>
              <td>replaced_by</td>
              <td>{returnNull(annotationsField.replaced_by)}</td>
            </tr>
            <tr className={returnClass(annotationsField.api_docs_url)}>
              <td>api_docs_url</td>
              <td>{returnNull(annotationsField.api_docs_url)}</td>
            </tr>
            <tr className={returnClass(annotationsField.for_wikis)}>
              <td>for_wikis</td>
              <td>{returnNull(annotationsField.for_wikis)}</td>
            </tr>
          </tbody>
          <tfoot className="explore--tfoot">
            <tr>
              <td colSpan="3">
                Fields With red background are missing!
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  );
}
