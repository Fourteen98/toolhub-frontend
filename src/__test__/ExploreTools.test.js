import React from 'react';
import renderer from 'react-test-renderer';
import ExploreTool from '../pages/components/ExploreTool';
import ToolsData from '../data/ToolsData';

describe('ExploreTool', () => {
  const tool = ToolsData.find((tool) => tool.id === Number(1));
  it('renders correctly', () => {
    const tree = renderer.create(<ExploreTool tool={tool} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
