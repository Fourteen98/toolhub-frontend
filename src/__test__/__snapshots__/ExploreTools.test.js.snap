// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreTool renders correctly 1`] = `
<div
  className="explore--tools--container"
>
  <img
    alt="Depictor"
    className="tool--card--img"
    src="Tool2.jpg"
  />
  <div
    className="left--container"
  >
    <h3
      className="tool--name"
    >
      Depictor
    </h3>
    <p>
      <span>
        Description: 
      </span>
      A tool for creating diagrams and flowcharts
    </p>
    <p>
      <span>
        Author: 
      </span>
      Damilare Adedoyin
    </p>
    <p>
      <span>
        Category: 
      </span>
      Diagramming
    </p>
    <p>
      <span>
        Source Code : 
      </span>
      <a
        href="https://depictor.app/"
        rel="noreferrer"
        target="_blank"
      >
         Check here 
      </a>
    </p>
  </div>
  <div
    className="right--container"
  >
    <table
      className="explore--tools--tbl"
    >
      <thead>
        <tr>
          <th>
            Field
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="missing"
        >
          <td>
            icon
          </td>
          <td>
            Missing!
          </td>
        </tr>
        <tr
          className="missing"
        >
          <td>
            feedback_url
          </td>
          <td>
            Missing!
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            license
          </td>
          <td>
            Apache License 2.0
          </td>
        </tr>
        <tr
          className="missing"
        >
          <td>
            privacy_policy
          </td>
          <td>
            Missing!
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            replaced_by
          </td>
          <td>
            https://www.draw.io/
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            api_docs_url
          </td>
          <td>
            https://www.draw.io/docs/
          </td>
        </tr>
        <tr
          className="missing"
        >
          <td>
            for_wikis
          </td>
          <td>
            Missing!
          </td>
        </tr>
      </tbody>
      <tfoot
        className="explore--tfoot"
      >
        <tr>
          <td
            colSpan="3"
          >
            NB: Fields with red background are missing!
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
